### PR TITLE
Updates to examples in fundamentals section

### PR DESCRIPTION
**Pull request by jdamiba**
The purpose of this PR is to update 3 of our examples.

### CODE REVIEW

File: doc/python/renderers.md [hunk 1]
```diff
@@ -223,12 +226,28 @@ fig = go.Figure(
...
-Dash is a Python framework for building web applications, and it provides built-in support for displaying Plotly figures. See the [Dash User Guide](https://dash.plot.ly/) for more information.
+### Displaying Figures Using Dash
+[Dash](https://dash.plot.ly) is a Python framework for building web applications, and it provides built-in support for displaying figures created with Plotly's graphing libraries. See the [Dash User Guide](https://dash.plot.ly/) for more information.
+
+It is important to note that Dash does not use the renderers framework discussed above, so you should not use the `show()` figure method or the `plotly.io.show` function inside Dash applications.
+
+Instead, pass your figure as the `figure` parameter to the `dcc.Graph` component. 
+
+```python
+import dash_core_components as dcc
```
Comment (emmanuelle) [inline doc/python/renderers.md]: You need to add `dash_core_components` to `doc/requirements.txt` and `binder/requirements.txt` (the CI fails at the moment).

File: doc/python/renderers.md [hunk 2]
```diff
@@ -223,12 +226,28 @@ fig = go.Figure(
...
-### Displaying figures using Dash
-Dash is a Python framework for building web applications, and it provides built-in support for displaying Plotly figures. See the [Dash User Guide](https://dash.plot.ly/) for more information.
+### Displaying Figures Using Dash
+[Dash](https://dash.plot.ly) is a Python framework for building web applications, and it provides built-in support for displaying figures created with Plotly's graphing libraries. See the [Dash User Guide](https://dash.plot.ly/) for more information.
+
+It is important to note that Dash does not use the renderers framework discussed above, so you should not use the `show()` figure method or the `plotly.io.show` function inside Dash applications.
+
+Instead, pass your figure as the `figure` parameter to the `dcc.Graph` component. 
+
+```python
```
Comment (nicolaskruchten) [inline doc/python/renderers.md]: does this actually work here? I would think you'd need to escape this block somehow so it doesn't actually execute :)

File: doc/python/subplots.md [hunk 3]
```diff
@@ -37,9 +37,11 @@ jupyter:
 
 #### Simple Subplot
 
-Figures with subplots are created using the `make_subplots` function from the `plotly.subplots` module.
+Figures with subplots are created using the `make_subplots()` function from the `plotly.subplots` module. 
 
-Here is an example of creating a figure with two scatter traces in side-by-side subplots.
+This function accepts two integer parameters (`rows` and `cols`), which specifiy how the subplots should be laid out.
```
Comment (nicolaskruchten) [inline doc/python/subplots.md]: this is probably misleading, as this function accepts many other parameters as well...

File: doc/python/subplots.md [hunk 4]
```diff
@@ -203,9 +205,9 @@ fig.show()
 ```
 
 #### Customizing Subplot Axes
-After a figure with subplots is created using the `make_subplots` function, its axis properties (title, font, range, grid style, etc.) can be customized using the `update_xaxes` and `update_yaxes` graph object figure methods.  By default, these methods apply to all of the x axes or y axes in the figure. The `row` and `col` arguments can be used to control which axes are targeted by the update.
+After a figure with subplots is created using the `make_subplots()` function, its axis properties (title, font, range, grid style, etc.) can be customized using the `update_xaxes` and `update_yaxes` graph object figure methods.  By default, these methods apply to all of the `x` axes or `y` axes in the figure. The `row` and `col` arguments can be used to control which axes are targeted by the update.
```
Comment (nicolaskruchten) [inline doc/python/subplots.md]: why put `x` and `y` in backticks here?

File: doc/python/subplots.md [hunk 5]
```diff
@@ -241,17 +243,17 @@ fig.show()
...
+Here is an example that creates a figure with 3 vertically stacked subplots with linked `x` axes. A small vertical spacing value is used to reduce the spacing between subplot rows.
 
 ```python
 from plotly.subplots import make_subplots
 import plotly.graph_objects as go
 
-fig = make_subplots(
-    rows=3, cols=1, shared_xaxes=True, vertical_spacing=0.02
-)
+fig = make_subplots(rows=3, cols=1,
```
Comment (nicolaskruchten) [inline doc/python/subplots.md]: this feels like a one-liner to me

File: doc/python/subplots.md [hunk 6]
```diff
@@ -354,7 +355,7 @@ fig.show()
 ```
 
 #### Subplots Types
-By default, the `make_subplots` function assumes that the traces that will be added to all subplots are 2-dimensional cartesian traces (e.g. `scatter`, `bar`, `histogram`, `violin`, etc.).  Traces with other subplot types (e.g. `scatterpolar`, `scattergeo`, `parcoords`, etc.) are supporteed by specifying the `type` subplot option in the `specs` argument to `make_subplots`.
+By default, the `make_subplots()` function assumes that the traces that will be added to all subplots are 2-dimensional cartesian traces (e.g. `scatter`, `bar`, `histogram`, `violin`, etc.).  Traces with other subplot types (e.g. `scatterpolar`, `scattergeo`, `parcoords`, etc.) are supporteed by specifying the `type` subplot option in the `specs` argument to `make_subplots`.
```
Comment (nicolaskruchten) [inline doc/python/subplots.md]: in general, I don't really agree with putting the `()` after the function name everywhere (this comment applies to this whole PR btw). The function is properly named `make_subplots` and calling it with no parameters like `make_subplots()` will probably result in an error. It's probably OK to use this for something like "the `.show()` method" as that actually will work with no arguments.

Also note: maybe we haven't used it super consistently, but the leading `.` in "the `.show()` method" is actually informative IMO in that it reminds the reader that `show` isn't a free function but a method of a figure. As a matter of style we could say "the `fig.show` method" or "the `.show` method of every `go.Figure` object" or something.

In any case, unless you're going to fully standardize the usage of this kind of thing throughout the docs, I find it pretty distracting and hard to review changes like this :)

File: doc/python/renderers.md [hunk 7]
```diff
@@ -171,18 +175,17 @@ fig.show(renderer="png")
 ```
 
 ###### `pdf`
-This renderer displays figures as static PDF files. This is especially useful for notebooks that will be exported to PDF files using the LaTeX export capabilities of nbconvert.
+This renderer displays figures as static `.pdf` files. This is especially useful for notebooks that will be exported to `.pdf` files using the `LaTeX` export capabilities of `nbconvert`.
```
Comment (nicolaskruchten) [inline doc/python/renderers.md]: here the usage of backticks is questionable also... "static PDF files" is acceptable, no? It's not clear why "static `.pdf` files" is clearer or better :)

File: doc/python/renderers.md [hunk 8]
```diff
@@ -106,23 +110,23 @@ fig = go.Figure(
...
 
-This renderer is a good choice for notebooks that will be exported to HTML files (Either using [nbconvert](https://nbconvert.readthedocs.io/en/latest/) or the "Download as HTML" menu action) because the exported HTML files will work without an internet connection.
+This renderer is a good choice for notebooks that will be exported to HTML files (Either using [nbconvert](https://nbconvert.readthedocs.io/en/latest/) or the "Download as HTML" menu action) because the exported HTML files will work without an Internet connection.
 
-> Note: Adding the Plotly.js bundle to the notebook does add a few megabytes to the notebook size, so if you can count on having an internet connection you may want to consider the `notebook_connected` renderer.
+> Note: Adding the `plotly.js` bundle to the notebook adds a few megabytes to the notebook size. If you can count on always having an Internet connection, you may want to consider using the `notebook_connected` renderer if notebook size is a constraint.
```
Comment (nicolaskruchten) [inline doc/python/renderers.md]: I don't think we should backtick "plotly.js" ...

File: doc/python/renderers.md [hunk 9]
```diff
@@ -33,19 +33,21 @@ jupyter:
...
-### Displaying plotly figures
-This section covers the many ways to display plotly figures from Python.  At the highest level, there are three general approaches:
+Plotly's Python graphing library, `plotly.py`, gives you a wide range of options for how and where to display your figures. 
 
- 1. Using the renderers framework in the context of a script or notebook
- 2. Using Dash in a web app context
- 3. Using a `FigureWidget` in an ipywidgets context
+In general, there are three different approaches you can take in order to display figures:
+
+ 1. Using the `renderers` framework in the context of a script or notebook
```
Comment (nicolaskruchten) [inline doc/python/renderers.md]: Here also I think the usage of backticks for `renderers` is going a bit too far IMO...

File: doc/python/creating-and-updating-figures.md [hunk 10]
```diff
@@ -487,34 +559,48 @@ fig.for_each_trace(
...
 ```
 
-There are also `for_each_xaxis` and `for_each_yaxis` methods that are analogous to the `for_each_trace` method described above. For non-cartesian subplot types (e.g. polar), there are additional `update_{type}` and `for_each_{type}` methods (e.g. `update_polar`, `for_each_polar`).
+There are also `for_each_xaxis()` and `for_each_yaxis()` methods that are analogous to the `for_each_trace()` method described above. For non-cartesian subplot types (e.g. polar), there are additional `update_{type}` and `for_each_{type}` methods (e.g. `update_polar()`, `for_each_polar()`).
 
-### Other update methods
+### Other Update Methods
 
-`go` figures also support `update_layout_images` in order to [update background layout images](/python/images/), `update_annotations` in order to [update annotations](/python/text-and-annotations/#multiple-annotations), and `update-shapes` in order to [update shapes](/python/shapes/).
+Figures created with the `plotly.graph_objects` module also support:
```
Comment (nicolaskruchten) [inline doc/python/creating-and-updating-figures.md]: This isn't totally correct: all figures support this, including those created with make_subplots and PX...

**Comment by nicolaskruchten** [comment]
As a general comment, I think the style rule around backticks could be something like "text that might reasonably end up being executed gets backticks"... this excludes `LaTeX` and "`x` axis" but includes `nbconvert` and `.show` and `go.Figure` and `px.scatter` ...  thoughts?

**Comment by jdamiba** [comment]
@nicolaskruchten I agree with your style comments and think this is a good opportunity to standardize. 

To summarize:

`fig`
`fig.show()`
`.show()`
`make_subplots`

No backticks for file formats. 

What about package and library names like `plotly.py`, `plotly.js` or `htmlwidgets`? I think they should get backticks.

**Comment by jdamiba** [comment]
@nicolaskruchten Can you please take another look at this.

File: doc/requirements.txt [hunk 11]
```diff
@@ -23,5 +23,4 @@ recommonmark
 pathlib
 python-frontmatter
 datashader
-pyarrow
-cufflinks==0.17.3
+pyarrow
```
Comment (nicolaskruchten) [inline doc/requirements.txt:26]: this diff should not be in this PR: we need cufflinks installed

Comment (nicolaskruchten) [reply doc/requirements.txt:26]: I'll make the change and merge this

**Pull request merged**